test-utils/src/cockroach_manager.rs

```diff
@@ -58,23 +66,70 @@ impl<'a> CockroachManager<'a> {
             &format!("--listen-addr=127.0.0.1:{}", self.port),
             &format!("--sql-addr=127.0.0.1:{}", self.sql_port),
             &format!("--http-addr={}", self.http_address),
-            &format!("--store={}", self.manager.storage_dir.as_path().display()),
             &format!("--certs-dir={}", self.certs_dir),
+            "--insecure",
+            "--store=type=mem,size=0.25",
+            // NOTE
+            // Uncomment it for debugging , there is not good reason always log
+            // Will be with #619
+            // &format!(r#"--log="{{file-defaults:{{dir:{}}},sinks:{{stderr:{{filter: NONE}}}}}}""#, self._output_dir.path().join("db.log").display())
         ];
-        self.manager.start("cockroach", args, check).await
-    }
 
-    pub fn kill(&mut self) {
-        self.manager.kill()
+        Command::new("cockroach")
+            .args(args)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .with_context(|| "failed to start cockroach".to_string())
     }
 }
 
-pub struct CockroachCheck(pub Settings);
+pub struct CockroachManager<'a> {
+    process: Child,
+    _output_dir: &'a TempDir,
+    pub sql_port: u16,
+}
 
-#[async_trait]
-impl Check for CockroachCheck {
-    async fn check(&self) -> bool {
-        connection(&self.0).await.is_ok()
+impl<'a> CockroachManager<'a> {
+    pub async fn builder(
```

Comment:
This is not really the proper use of the builder pattern. I think it would better to leave this as the new() function and then have a start() function instead of build().

Reply:
It is nothing about syntax, it is about memory when running.